ci-instability: curl: (7) Failed to connect to go.otr.dev port 443: Connection refused

**Describe the bug**

```
fetching cli artifacts
[2021-09-01T21:08:21Z] downloading https://go.otr.dev/cli-latest-release-linux
[2021-09-01T21:08:22Z]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[2021-09-01T21:08:22Z]                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to go.otr.dev port 443: Connection refused
[2021-09-01T21:08:22Z] make[1]: *** [Makefile:763: testupgrade-fetch-cli-artifacts] Error 7
[2021-09-01T21:08:22Z] make[1]: Leaving directory '/build'
[2021-09-01T21:08:22Z] make: *** [Makefile:755: test-upgrade-run-sh] Error 2
[2021-09-01T21:08:23Z] make: *** [ci-test-upgrade-run-sh] Error 2
```

https://buildkite.com/opstrace/scheduled-main-builds/builds/3010#600d81fc-c435-4e1b-a96c-9e091ceb62ea/262-266

--- a/ci/test-upgrade/fetch-cli-artifacts.sh
+++ b/ci/test-upgrade/fetch-cli-artifacts.sh
@@ -2,6 +2,8 @@
 
 set -eou pipefail
 
+source ci/utils.sh
+
 if [ -z ${OPSTRACE_CLI_VERSION_FROM+x} ];
 then
     echo "OPSTRACE_CLI_VERSION_FROM env var is not set"
@@ -20,7 +22,9 @@ fi
 #
 fetch_cli_s3_artifact() {
     echo "downloading ${artifact}"
-    aws s3 cp --only-show-errors ${artifact} .
+    # aws cli retry modes documentation
+    # https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html
+    AWS_MAX_ATTEMPTS=3 AWS_RETRY_MODE=standard aws s3 cp --only-show-errors ${artifact} .
 
     echo "extracting ${artifact} to target dir ${dir}"
     tar xjf $(basename ${artifact#s3://}) -C ${dir}
@@ -32,6 +36,8 @@ fetch_cli_s3_artifact() {
 #
 fetch_cli_http_artifact() {
     echo "downloading ${artifact}"
+    # curl retry flag documentation
+    # https://man7.org/linux/man-pages/man1/curl.1.html
     curl -L ${artifact} --output opstrace-cli-artifact.tar.bz2
 
     echo "extracting ${artifact} to target dir ${dir}"

